core/src/main/scala/org/apache/spark/executor/ProcfsMetricsGetter.scala

```diff
@@ -62,13 +64,14 @@ private[spark] class ProcfsMetricsGetter(procfsDir: String = "/proc/") extends L
         SparkEnv.get.conf.get(config.EVENT_LOG_STAGE_EXECUTOR_METRICS)
       val shouldLogStageExecutorProcessTreeMetrics =
         SparkEnv.get.conf.get(config.EVENT_LOG_PROCESS_TREE_METRICS)
-      procDirExists.get && shouldLogStageExecutorProcessTreeMetrics && shouldLogStageExecutorMetrics
+      procDirExists.get && shouldLogStageExecutorProcessTreeMetrics &&
+        shouldLogStageExecutorMetrics && Shell.LINUX
```

Comment:
Are we sure to support only Linux? Other Unix-like systems which have /proc?

Reply:
In hadoop ProcfsMetrics, it checks this [ProcfsBasedProcessTree](https://github.com/apache/hadoop/blob/2064ca015d1584263aac0cc20c60b925a3aff612/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/ProcfsBasedProcessTree.java#L177) and it already exists long time.

Reply:
Looks like in isProcfsAvailable, we check /proc existence instead of checking Linux like hadoop. Isn't it enough? And for other Unit-like systems which have /proc? We do not need to follow ProcfsBasedProcessTree.

Reply:
AFAIC the procfs is the linux feature, other unix-like os may be try to compatible with it. Check the  /proc dir is not a better idea than check linux os. One point is that, someone can change the /proc information which os not support procfs native.

Reply:
If /proc in other Unix-like systems, this change actually removes the support on these systems. Unless we are sure current code does not work for such systems, I think we should not simply remove their support.